GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java

```diff
@@ -58,7 +65,52 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
             purgeOrphanJobQueueRecords();
         } else if ("generateNotifications".equals(action)) {
             generateNotifications();
+        } else if ("purgeDeviceFileJobQueueRecords".equals(action)) {
+            purgeDeviceFileJobQueueRecords();
+        }
+    }
+
+    /**
+     * scans for and deletes DeviceFileJobQueue entries that are either more than two years old
+     * or that refer to files that have been successfully uploaded.
+     */
+    private void purgeDeviceFileJobQueueRecords() {
+        Calendar deadline = Calendar.getInstance();
+        deadline.add(Calendar.YEAR, -2); //two years ago
+        log.info("Starting scan for DFJQ entries, fulfilled or older than: " + deadline.getTime());
+        DeviceFileJobQueueDAO dfjqDao = new DeviceFileJobQueueDAO();
+        List<DeviceFileJobQueue> dfjqList = dfjqDao.list("all");
+        int retirees = 0;
+        for (DeviceFileJobQueue item : dfjqList) {
+            if (item.getCreatedDateTime() != null
+                    && deadline.getTime().after(item.getCreatedDateTime())) {
+                //cheap case - old
+                log.info("Deleting old DFJQ entry: " + item.getKey().getId());
+                SurveyTaskUtil.spawnDeleteTask(SurveyTaskRequest.DELETE_DFJQ_ACTION,
+                        item.getKey().getId());
+                retirees++;
+            } else {//check the (now protected)image file in S3 store - need credentials
+                try {
+                    String bucket = 
+                            com.gallatinsystems.common.util.PropertyUtil.getProperty("s3bucket");
+                    HttpURLConnection conn = (HttpURLConnection)
+                            S3Util.getConnection(bucket, "images/" + item.getFileName());
+                    log.info("Checking for " + item.getFileName() + 
+                            " : " + conn.getResponseCode() + " " + conn.getResponseMessage());
+                    if (conn.getResponseCode() == 200) {
+                        // best case - fulfilled
+                        log.info("Deleting fulfilled DFJQ entry: " + item.getKey().getId());
+                        SurveyTaskUtil.spawnDeleteTask(SurveyTaskRequest.DELETE_DFJQ_ACTION,
+                                item.getKey().getId());
+                        retirees++;
+                    }
+                } catch (Exception e) {
+                    // catchall
+                    log.warning("Error while connecing to " + item.getFileName() +"\n\n" + e.getMessage());
+                }
+            }
         }
+        log.info("Attempted to retire "+retirees+" of " + dfjqList.size());
```

Comment:
it seems to be there are spaces missing, shouldn't it be ...retire " + retirees + "